### PR TITLE
Remove the `cacheDirectory` option from babel config

### PR DESCRIPTION
**Pull request by tmeasday**
Issue: https://github.com/storybooks/storybook/issues/1254

## What I did

In storyshots, we are running babel directly, not through webpack, so we can't use this option, which applies to `babel-loader`. See https://github.com/storybooks/storybook/blob/master/app/react/src/server/config/babel.js#L15=

## How to test

Trusty ol' `test-cra`, running `npm test` from inside `examples/test-cra`.

### CODE REVIEW

File: addons/storyshots/src/index.js [hunk 1]
```diff
@@ -31,11 +31,14 @@ export default function testStorySnapshots(options = {}) {
...
     // eslint-disable-next-line
-    const loadBabelConfig = require('@storybook/react/dist/server/babel_config').default;
+    const loadBabelConfig = require('@storybook/react/dist/server/babel_config')
+      .default;
     const configDirPath = path.resolve(options.configPath || '.storybook');
     configPath = path.join(configDirPath, 'config.js');
 
     const babelConfig = loadBabelConfig(configDirPath);
+    // We set this in the default babel config, but it is webpack-only
+    delete babelConfig.cacheDirectory;
```
Comment (tmeasday) [inline addons/storyshots/src/index.js]: Potentially we should remove this config from this file: https://github.com/storybooks/storybook/blob/master/app/react/src/server/config/babel.js#L15= and instead add it when adding the config to webpack.

**Comment by tmeasday** [comment]
This issue would only surface if you ran `jest` in a directory that did not contain a `.babelrc`.

**Review by shilman — APPROVED** [review]
@tmeasday @ndelangen I don't actually understand this completely, so I'm going to approve because it makes sense, but I'll let @ndelangen approve/merge.

File: addons/storyshots/src/index.js (hunk 1 above)
Comment (ndelangen) [inline addons/storyshots/src/index.js]: Sounds better!

Comment (tmeasday) [inline addons/storyshots/src/index.js]: OK, done! Please take a look at the last commit.

Along the way this actually improves the behaviour in RN, before it wasn't using a cache dir if you had a custom `.babelrc` (which I reckon you *always* would in RN)

**Comment by shilman** [comment]
@ndelangen merging this with @tmeasday 's updates, which have been tested for both react and RN. going to do a small release to get this and other patch fixes out.

**Pull request merged**